torch/csrc/utils/tensor_new.cpp

```diff
@@ -169,6 +169,9 @@ ScalarType infer_scalar_type(PyObject *obj) {
   if (PyBool_Check(obj)) {
     return ScalarType::Bool;
   }
+  if (PyComplex_Check(obj)) {
+    return ScalarType::ComplexDouble;
```

Comment:
However, I am wondering; above we have `torch::tensors::get_default_scalar_type()`, so I wonder if complex should switch between ComplexDouble and ComplexFloat depending on the default scalar type. Something to think about.

Reply:
I think that's because those float and double exist in std library, so we can just easily infer the scalar types accordingly.

Reply:
I'm not sure what you mean by "exist in the std library"; `get_default_scalar_type` doesn't have much to do with `std`.